tests/imputers/test_ModeImputer.py

```diff
@@ -4,71 +4,46 @@
 import test_aide as ta
 
 import tests.test_data as d
-import tubular
+from tests.base_tests import (
+    ColumnStrListInitTests,
+    GenericFitTests,
+    GenericTransformTests,
+    OtherBaseBehaviourTests,
+)
+from tests.imputers.test_BaseImputer import GenericImputerTransformTests
 from tubular.imputers import ModeImputer
 
 
-class TestInit:
-    """Tests for ModeImputer.init()."""
+class TestInit(ColumnStrListInitTests):
+    """Generic tests for transformer.init()."""
 
-    def test_super_init_called(self, mocker):
-        """Test that init calls BaseTransformer.init."""
-        expected_call_args = {
-            0: {"args": (), "kwargs": {"columns": None, "verbose": True}},
-        }
+    @classmethod
+    def setup_class(cls):
+        cls.transformer_name = "ModeImputer"
 
-        with ta.functions.assert_function_call(
-            mocker,
-            tubular.base.BaseTransformer,
-            "__init__",
-            expected_call_args,
-        ):
-            ModeImputer(columns=None, verbose=True)
+    def test_weight_value_type_error(
```

Comment:
There's another [issue](https://github.com/lvgig/tubular/issues/221) re updating handling of weights, so can leave this in for the time being

Reply:
No probs